packages/terra-clinical-item-collection/src/Item.jsx

```diff
@@ -0,0 +1,151 @@
+  /**
+   * Whether or not the item is selectable. If true, the item is given list and table hover and focus styles and set tabIndex
+   * to 0.
+   */
+  isSelectable: PropTypes.bool,
+  /*
+   * Whether or not the item is selected.
+   */
+  isSelected: PropTypes.bool,
+  /**
+   * The view in which the item should be presented. Options are list or table and this will be set by the Item Collection component.
+   */
+  view: PropTypes.oneOf(['list', 'table']),
+};
+
+const defaultProps = {
+  startAccessory: undefined,
+  comment: undefined,
+  endAccessory: undefined,
+  accessoryAlignment: 'alignCenter',
+  isListItemTruncated: false,
+  isSelectable: false,
+  isSelected: false,
+  listItemLayout: 'oneColumn',
+  listItemTextEmphasis: 'default',
+  view: 'list',
+};
+
+function createListItem(elements, selectableProps, isSelected, itemViewStyles) {
+  const { startAccessory, children, comment, endAccessory, hasStartAccessory, hasEndAccessory, ...listItemProps } = elements;
+
+  const listItemContent = (
+    <ItemView
+      startAccessory={startAccessory}
+      displays={React.Children.toArray(children)}
+      comment={comment}
+      endAccessory={endAccessory}
+      reserveStartAccessorySpace={hasStartAccessory}
+      {...itemViewStyles}
+    />
+  );
+
+  return (
+    <List.Item
+      content={listItemContent}
+      isSelected={isSelected}
+      {...selectableProps}
+      {...listItemProps}
+    />
+  );
+}
+
+function createTableCell(content, keyValue, contentType, accessoryAlignment) {
+  const cellClassNames = cx(
+    `content-${contentType}`,
+    { 'content-accessory-align-center': (contentType.includes('accessory') && accessoryAlignment === 'alignCenter') },
+    { 'content-accessory-align-top': (contentType.includes('accessory') && accessoryAlignment === 'alignTop') },
+  );
+
+  return (<Table.Cell content={content} key={keyValue} className={cellClassNames} />);
+}
+
+function createTableRow(elements, selectableProps, isSelected, accessoryAlignment) {
+  const { startAccessory, children, comment, endAccessory, ...tableRowProps } = elements;
+
+  const displayContent = React.Children.map(children, (display, index) => {
+    const displayKey = `display_${index + 1}`;
+    return createTableCell(display, displayKey, 'display');
+  });
+
+  return (
+    <Table.Row isSelected={isSelected} {...selectableProps} {...tableRowProps}>
+      {startAccessory && createTableCell(startAccessory, 'start_accessory', 'start-accessory', accessoryAlignment)}
+      {displayContent}
+      {comment && createTableCell(comment, 'comment', 'comment')}
+      {endAccessory && createTableCell(endAccessory, 'end_accessory', 'end-accessory', accessoryAlignment)}
+    </Table.Row>
+  );
+}
+
+const Item = (props) => {
+  const {
+    accessoryAlignment,
+    isListItemTruncated,
+    isSelectable,
+    isSelected,
+    listItemLayout,
+    listItemTextEmphasis,
+    view,
+    ...elements
```

Comment:
Is `elements` here representing `customProps` or `children`? I think we should still spread custom props onto the TableRow / ItemView, or at the minimum allow an onClick to be passed through. 

Each ItemCollection.Item should still be able to respond to an onClick, is that possible with the current implementation?

Reply:
elements includes children, startAccessory, displays, endAccessory and customProps. Custom props are broken out [here](https://github.com/cerner/terra-clinical/pull/221/files/53645d2119839ea08d1ec8912d7aec688a7f305d#diff-93e672701138c7e816ff34f0d5b7dc53R74) & [here](https://github.com/cerner/terra-clinical/pull/221/files/53645d2119839ea08d1ec8912d7aec688a7f305d#diff-93e672701138c7e816ff34f0d5b7dc53R108), so custom props are spread on the list items and table rows. This should probably be cleaned up to make this clearer.